dlc-manager/src/contract_updater.rs

```diff
@@ -565,63 +588,67 @@ where
         )?;
     }
 
-    let mut input_serials: Vec<_> = offered_contract
+    let fund_tx = &accepted_contract.dlc_transactions.fund;
+    let mut unsigned_tx = fund_tx.clone();
+    // need to remove sigs from fund_tx for PSBT
+    unsigned_tx.input.iter_mut().for_each(|x| {
+        x.script_sig = Script::new();
+        x.witness.clear();
+    });
+    let mut fund_psbt = PartiallySignedTransaction::from_unsigned_tx(unsigned_tx).unwrap();
+    // add sigs to fund_psbt
+    for i in 0..fund_psbt.inputs.len() {
+        fund_psbt.inputs[i].final_script_sig = Some(fund_tx.input[i].script_sig.clone());
+        fund_psbt.inputs[i].final_script_witness = Some(fund_tx.input[i].witness.clone());
+    }
```

Comment:
That doesn't feel like the way to go tbh. I think you just need to make sure that the contracts only keep unsigned transactions (I can have a look if you want let me know).

Reply:
Yeah this felt kinda weird here. Wasn't sure if the funding sigs were only stored in this tx.

Reply:
They're stored in the `SignedContract` which I think is enough.